modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -513,13 +519,38 @@ private CompletableFuture<?> processRequest(ReplicaRequest request, @Nullable Bo
             return processGetEstimatedSizeRequest();
         }
 
-        HybridTimestamp opTsIfDirectRo = (request instanceof ReadOnlyDirectReplicaRequest) ? clockService.now() : null;
+        @Nullable HybridTimestamp opTs = getTxOpTimestamp(request);
+        @Nullable HybridTimestamp opTsIfDirectRo = (request instanceof ReadOnlyDirectReplicaRequest) ? opTs : null;
+        @Nullable HybridTimestamp txTs = getTxStartTimestamp(request);
+        if (txTs == null) {
```

Comment:
It seems safer to switch to `request instanceof ReadOnlyDirectReplicaRequest` condition (or at least assert it)

Reply:
There is no need in such check or assertion.
txTs can be assigned to non null value only if request instanceof ReadOnlyDirectReplicaRequest

Reply:
In the current code this is true, but if we change the code, we can inadvertently break this implicit invariant. An assertion would help to make it explicit.